samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/QuestionPoolFacadeQueries.java

```diff
@@ -1417,11 +1417,18 @@ public Object doInHibernate(Session session) throws HibernateException, SQLExcep
 	  return count;
   }
   
+  /**
+   * Fetch a HashMap of question pool ids and counts for all pools that a user has access to.
+   * We inner join the QuestionPoolAccessData table because the user may have access to pools
+   * that are being shared by other users. We can't simply look for the ownerId on QuestionPoolData.
+   * This was originally written for SAM-2463 to speed up these counts. 
+   * @param agentId Sakai internal user id. Most likely the currently logged in user
+   */
   public HashMap<Long, Integer> getCountItemFacadesForUser(final String agentId) {	    
 	  final HibernateCallback hcb = new HibernateCallback(){
 		  public Object doInHibernate(Session session) throws HibernateException, SQLException {
-			  Query q = session.createQuery("select qpi.questionPoolId, count(ab) from ItemData ab, QuestionPoolItemData qpi, QuestionPoolData qpd " + 
-					  "where ab.itemId=qpi.itemId and qpi.questionPoolId=qpd.questionPoolId AND qpd.ownerId=? group by qpi.questionPoolId");
+			  Query q = session.createQuery("select qpi.questionPoolId, count(ab) from ItemData ab, QuestionPoolItemData qpi, QuestionPoolData qpd, QuestionPoolAccessData qpad " + 
+					  "where ab.itemId=qpi.itemId and qpi.questionPoolId=qpd.questionPoolId AND qpd.questionPoolId=qpad.questionPoolId AND qpad.agentId=? group by qpi.questionPoolId");
 			  q.setString(0, agentId);
```

Comment:
I believe that it should be checked that the logged in user has allowed access to the pool:

![image](https://cloud.githubusercontent.com/assets/10761548/6665606/b7974f06-cbdb-11e4-81e9-867fdb656a65.png)